app/src/main/java/com/android/academy/fundamentals/workshop_1/solution/WS01ThreadsSolutionFragment.kt

```diff
@@ -0,0 +1,83 @@
+package com.android.academy.fundamentals.workshop_1.solution
+
+import android.os.*
+import android.view.View
+import android.widget.Button
+import android.widget.TextView
+import androidx.fragment.app.Fragment
+import com.android.academy.fundamentals.R
+
+class WS01ThreadsSolutionFragment: Fragment(R.layout.fragment_theads) {
+
+    private var threadButton : Button? = null
+    private var threadTextView : TextView? = null
+
+    private val handler = MyHandler()
+
+    override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
+        findViews(view)
+        threadButton?.setOnClickListener {
+            startThread()
+            startRunnable()
+        }
+    }
+
+    override fun onDestroyView() {
+        threadButton = null
+        threadTextView = null
+        super.onDestroyView()
+    }
+
+    private fun findViews(view: View) {
+        threadButton = view.findViewById(R.id.thread_button)
+        threadTextView = view.findViewById(R.id.thread_text_view)
+    }
+
+    private fun startThread() {
+        val thread = MyThread()
+        thread.start()
+        printMessage(getString(R.string.wait))
+    }
+
+    private fun startRunnable() {
+        val thread = Thread(MyRunnable())
+        thread.start()
+        printMessage(getString(R.string.wait))
+    }
+
+    private fun printMessage(mes: String){
+        threadTextView?.text =  mes
+    }
+
+    inner class MyThread : Thread() {
```

Comment:
qst: is there any point to declare class as inner class? As I remember we didn't explain this point and it is not important for students